precompiles/staking/tx.go

```diff
@@ -81,14 +84,65 @@ func (p Precompile) CreateValidator(
 		return nil, err
 	}
 
-	// Emit the event for the delegate transaction
+	// Emit the event for the create validator transaction
 	if err = p.EmitCreateValidatorEvent(ctx, stateDB, msg, delegatorHexAddr); err != nil {
 		return nil, err
 	}
 
 	return method.Outputs.Pack(true)
 }
 
+// EditValidator performs edit validator.
+func (p Precompile) EditValidator(
+	ctx sdk.Context,
+	origin common.Address,
+	_ *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, validatorHexAddr, err := NewMsgEditValidator(args)
+	if err != nil {
+		return nil, err
+	}
+
+	commissionRate := strconv.Itoa(DoNotModifyCommissionRate)
+	if msg.CommissionRate != nil {
+		commissionRate = msg.CommissionRate.String()
+	}
+
+	minSelfDelegation := strconv.Itoa(DoNotModifyMinSelfDelegation)
```

Comment:
why is this not defined as string in the first place?

Reply:
You can refer to the **Some Design Details** section in the PR I mentioned. Additionally, I would like to add that the reason for using int256 for the input parameters `commissionRate` and `minSelfDelegation` instead of using string is because in the already implemented StakingI.sol file, the Dec and Int types from Go language are both represented as uint256 in Solidity. I think this is a good design choice.

Reply:
@Vvaradinov What do you think of this design?

Reply:
Using it as a sentinel value makes sense to me but here you are using it only for the debug message ?

Reply:
the conversion negatively affects performance and is unnecessary. You don't need to convert

Reply:
> the conversion negatively affects performance and is unnecessary. You don't need to convert

Do you mean to define DoNotModifyMinSelfDelegation = "1" instead of DoNotModifyMinSelfDelegation = 1, or define DoNotModifyMinSelfDelegation = "[do-not-modify]"

Reply:
> `DoNotModifyMinSelfDelegation = "[do-not-modify]"`

This needs to be a constant

Reply:
Please address this also